add multi-tenancy public tutorial (#2393)
giantswarm/docs: src/content/tutorials/observability/multi-tenancy/creating-grafana-organization/_index.md

---
linkTitle: Creating a Grafana organization
title: Creating a Grafana organization
description: Guide explaining how to manage Grafana organizations in the Observability Platform.
menu:
  principal:
    identifier: tutorials-observability-multitenancy-create-grafana-organization
    parent: tutorials-observability-multitenancy
weight: 40
last_review_date: 2024-12-12
user_questions:
  - How to create a grafana organization?
  - How to access multi-tenant observability data?
owner:
  - https://github.com/orgs/giantswarm/teams/team-atlas
---

> When you first access [your installations `Grafana`]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}), you will be accessing an organization named _Shared Org_, which gives you access to a set of curated dashboards. However, if you want to provide multi-tenancy of your observability data (isolation between tenants), the observability platform allows you to create your own organizations in self-service.

Reviewer: If it's not already written / planned for another resource, it would be good to include a note about what multi-tenancy actually is in this context. e.g.
```suggestion
Grafana Organizations provide logical separation of dashboards, annotations, and other Grafana resources shown in the Grafana UI. When initially [accessing an installation's Grafana instance]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}), the default _Shared_ organization provides access to a set of common, curated dashboards visible to all users. The observability platform allows self-service creation of additional organizations to provide more a more tailored experience for platform users and more granular control for platform administrators.

Note: data from all organizations remains stored locally in the management cluster. Creating a Grafana Organization should be considered a privileged operation, since it allows access to all log and metric data in the MC.
```

## Creating your own organization

To create your own `Grafana` organization, you can create a [`GrafanaOrganization`](https://raw.githubusercontent.com/giantswarm/observability-operator/refs/heads/main/config/crd/observability.giantswarm.io_grafanaorganizations.yaml) custom resource in the management cluster containing the `Grafana` organization specification.

For example:

```yaml
apiVersion: observability.giantswarm.io/v1alpha1
kind: GrafanaOrganization
metadata:
  name: giantswarm
spec:
  displayName: Giant Swarm
  rbac:
    admins:
    - giantswarm-admins
    editors:
    - giantswarm-editors
    viewers:
    - giantswarm-viewers
  tenants:
  - giantswarm
```

Our operators will create this `Grafana` organization named _Giant Swarm_. It will be equipped with a basic set of data sources for Loki, Mimir and Alertmanager, giving you access to the `giantswarm` tenant.

> The Role Base Access Control (RBAC) section defines how to map your groups from your identity provider to `Grafana` admin roles (`Admin`, `Editor`, `Viewer`). Note that only the `admins` field is mandatory in this section.

Reviewer 2: Can you provide more information and examples regarding RBAC like, where does those values comes from. Probably adding a link to Grafana OAuth would be helpfull too

Reviewer: +1 references would be helpful, and beyond the scope of this PR, it might be nice to also include a reference to this doc in other OIDC RBAC docs to give admins a clue about this when they are doing group management

Author: This is actually a good idea

Author: I thinks this should be better now, What do you think?

---
linkTitle: Creating a Grafana organization
title: Creating a Grafana organization
description: Guide explaining how to manage Grafana organizations in the Observability Platform.
menu:
  principal:
    identifier: tutorials-observability-multitenancy-create-grafana-organization
    parent: tutorials-observability-multitenancy
weight: 40
last_review_date: 2024-12-12
user_questions:
  - How to create a grafana organization?
  - How to access multi-tenant observability data?
owner:
  - https://github.com/orgs/giantswarm/teams/team-atlas
---

> When you first access [your installations Grafana]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}), you will be accessing an organization named _Shared Org_ which gives you access to a set of curated dashboards. However, if you want to be able to provide multi-tenancy of your observability data (data isolation between tenants), the Observability Platform allows you to create your own organizations in self-service.

Reviewer 3: ```suggestion
When you first access [your installations `Grafana`]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}), you will be accessing an organization named _Shared Org_, which gives you access to a set of curated dashboards. However, if you want to provide multi-tenancy of your observability data (isolation between tenants), the observability platform allows you to create your own organizations in self-service.
```

## Creating your own organization

> To create your own Grafana organization, you can create a [`GrafanaOrganization`](https://raw.githubusercontent.com/giantswarm/observability-operator/refs/heads/main/config/crd/observability.giantswarm.io_grafanaorganizations.yaml) custom resource in the management cluster containing the Grafana organization specification.

Reviewer 3: ```suggestion
To create your own `Grafana` organization, you can create a [`GrafanaOrganization`](https://raw.githubusercontent.com/giantswarm/observability-operator/refs/heads/main/config/crd/observability.giantswarm.io_grafanaorganizations.yaml) custom resource in the management cluster containing the `Grafana` organization specification.
```

Reviewer: I don't think Grafana should be in a code block here

Reviewer 3: yeah this is a bit controversial. Or we use lowercase without quotes or we use quotes and uppercase. (Following MS style guide). Now, we have used quotes for project names. I am not proud but I want to unify it, we can change it though but not in this PR please

Reviewer: Grafana is not a technical concept here, it's a trademarked name of a product. Just as Kubernetes is [not code-blocked in Kubernetes documentation](https://kubernetes.io/docs/contribute/style/style-guide/) (and shouldn't be in ours), Grafana needs no special handling here IMO, also [according to MS](https://learn.microsoft.com/en-us/style-guide/text-formatting/formatting-common-text-elements).

Reviewer 3: hmm, what I meant is our guidelines encourage [lower case nouns](https://intranet.giantswarm.io/docs/content/style-guide/#capitalization) so I tend to use lowercase. In case of project/tools I have used title plus backtick. In MS guide style opt for title case but they dont say anything about quoting. The decision of quoting is highlight the tool (we did one introducing it firs time and I found contradictory not doing it always. But I may be too strict on that, I happy to remove backticks in followup PR (you are not the first asking for it)

> To create your own grafana organization, you can create a [`GrafanaOrganization`](https://raw.githubusercontent.com/giantswarm/observability-operator/refs/heads/main/config/crd/observability.giantswarm.io_grafanaorganizations.yaml) custom resource in the management cluster containing the Grafana organization specification. 

Author: ```suggestion
To create your own Grafana organization, you can create a [`GrafanaOrganization`](https://raw.githubusercontent.com/giantswarm/observability-operator/refs/heads/main/config/crd/observability.giantswarm.io_grafanaorganizations.yaml) custom resource in the management cluster containing the Grafana organization specification.
```

For example:

```yaml
apiVersion: observability.giantswarm.io/v1alpha1
kind: GrafanaOrganization
metadata:
  name: giantswarm
spec:
  displayName: Giant Swarm
  rbac:
    admins:
    - giantswarm-admins
    editors:
    - giantswarm-editors
    viewers:
    - giantswarm-viewers
  tenants:
  - giantswarm
```

> This Grafana organization will be created by our operators with the name *Giant Swarm*.

Author: ```suggestion
This Grafana organization will be created by our operators with the name _Giant Swarm_.
```

> It will be equipped with a basic set of datasources for Loki, Mimir and Alertmanager giving you access to the *giantswarm* tenant.

Author: ```suggestion
It will be equipped with a basic set of datasources for Loki, Mimir and Alertmanager giving you access to the _giantswarm_ tenant.
```

> This Grafana organization will be created by our operators with the name _Giant Swarm_.
> It will be equipped with a basic set of datasources for Loki, Mimir and Alertmanager giving you access to the _giantswarm_ tenant.

Reviewer 3: ```suggestion
Our operators will create this `Grafana` organization named _Giant Swarm_. It will be equipped with a basic set of data sources for Loki, Mimir and Alertmanager, giving you access to the `giantswarm` tenant.
```

Reviewer: Same here. `giantswarm` yes, `Grafana` no

> The RBAC section defines how to map your sso groups to grafana admin roles (Admin, Editor, Viewer). Note that only admins is mandatory in this section.

Reviewer 3: ```suggestion
The Role Base Access Control (RBAC) section defines how to map your groups from your identity provider to `Grafana` admin roles (`Admin`, `Editor`, `Viewer`). Note that only the `admins` field is mandatory in this section.
```

---
linkTitle: Creating a Grafana organization
title: Creating a Grafana organization
description: Guide explaining how to manage Grafana organizations in the Observability Platform.
menu:
  principal:
    identifier: tutorials-observability-multitenancy-create-grafana-organization
    parent: tutorials-observability-multitenancy
weight: 40
last_review_date: 2024-12-12
user_questions:
  - How to create a grafana organization?
  - How to access multi-tenant observability data?
owner:
  - https://github.com/orgs/giantswarm/teams/team-atlas
---

> When you first access [your installation's `Grafana`]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}), you will be accessing an organization named _Shared Org_, which gives you access to a set of curated dashboards. However, if you want to provide multi-tenancy of your observability data (isolation between tenants), the observability platform allows self-service creation of your own organizations.

Reviewer 4: ```suggestion
At the initial log in to [your installation's `Grafana`]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}) the preselected `Grafana` organization is the so called `_Shared Org_`. This shared organization contains a curated set of managed dashboards that are accessible to everyone with access to `Grafana`. If multiple teams access the observability platform we recommend to work with multi-tenancy and the resulting isolation of data and dashboards. For this use case the observability platform allows you to create new organizations in self-service. 
```

That's a bit heavy repition on the "access" here.

> At the initial log in to [your installation's `Grafana`]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}) the preselected `Grafana` organization is the so called `_Shared Org_`. This shared organization contains a curated set of managed dashboards that are accessible to everyone with access to `Grafana`. If multiple teams access the observability platform we recommend to work with multi-tenancy and the resulting isolation of data and dashboards. For this use case the observability platform allows you to create new organizations in self-service. 

Author: ```suggestion
At the initial log in to [your installation's `Grafana`]({{< relref "/tutorials/observability/data-exploration/accessing-grafana" >}}) the preselected `Grafana` organization is the so called `_Shared Org_`. This shared organization contains a curated set of managed dashboards that are accessible to everyone with access to `Grafana`. If multiple teams access the observability platform we recommend to work with multi-tenancy and the resulting isolation of data and dashboards. For this use case the observability platform allows you to create new organizations in self-service.
```

## Creating your own organization

To add a new `Grafana` organization, create a [`GrafanaOrganization`](https://raw.githubusercontent.com/giantswarm/observability-operator/refs/heads/main/config/crd/observability.giantswarm.io_grafanaorganizations.yaml) custom resource in the management cluster.

For example:

```yaml
apiVersion: observability.giantswarm.io/v1alpha1
kind: GrafanaOrganization
metadata:
  name: giantswarm
spec:
  displayName: Giant Swarm
  rbac:
    admins:
    - organization-admins
    editors:
    - dashboards-maintainers
    viewers:
    - data-readers
  tenants:
  - giantswarm
```

Our operators will create this `Grafana` organization named _Giant Swarm_. It will be equipped with a basic set of data sources for Loki, Mimir and Alertmanager, giving you access to the `giantswarm` tenant.

> The Role Base Access Control (RBAC) section defines how to assign groups from your configured identity provider to `Grafana` [available roles](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/#organization-roles) (`Admin`, `Editor`, `Viewer`). For organization mapping, you can read the official Grafana [documentation](https://grafana.com/docs/grafana/next/setup-grafana/configure-security/configure-authentication/generic-oauth/#configure-role-mapping).

Reviewer 5: Maybe it could help providing a "concrete" example here on how a real-life org would look like with an actual identity provider, but that's mostly nitpicking.

Author: The one above is the best thing I could find as a concrete example but I'm open to ideas

Reviewer 4: I think the example is quite okay. We could change it to something non-Giant Swarm like: 
Org/Tenant Name: MyOnlineShop
Admins = PlatformTeam
Editors = DevelopmentTeam
Viewers = MarketingTeam

Would that be more concrete for you @QuantumEnigmaa ?

  displayName: Giant Swarm
  rbac:
    admins:
    - organization-admins
    editors:
    - dashboards-maintainers
    viewers:
    - data-readers
  tenants:
  - giantswarm
```

Our operators will create this `Grafana` organization named _Giant Swarm_. It will be equipped with a basic set of data sources for Loki, Mimir and Alertmanager, giving you access to the `giantswarm` tenant.

> The Role Base Access Control (RBAC) section defines how to assign groups from your configured identity provider to `Grafana` [predefined roles](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/#organization-roles) (`Admin`, `Editor`, `Viewer`). organization mapping, you can read the official Grafana [documentation](https://grafana.com/docs/grafana/next/setup-grafana/configure-security/configure-authentication/generic-oauth/#configure-role-mapping).

Reviewer 2: I think in the open source version of Grafana we cannot add more roles.
```suggestion
The Role Base Access Control (RBAC) section defines how to assign groups from your configured identity provider to `Grafana` [available roles](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/#organization-roles) (`Admin`, `Editor`, `Viewer`). For organization mapping, you can read the official Grafana [documentation](https://grafana.com/docs/grafana/next/setup-grafana/configure-security/configure-authentication/generic-oauth/#configure-role-mapping).
```
